spring-tx/src/main/java/org/springframework/transaction/reactive/TransactionalOperatorImpl.java

```diff
@@ -71,6 +71,21 @@ public ReactiveTransactionManager getTransactionManager() {
 		return this.transactionManager;
 	}
 
+	@Override
+	public <T> Mono<T> transactional(Mono<T> mono) {
+		return TransactionContextManager.currentContext().flatMap(context -> {
+			Mono<ReactiveTransaction> status = this.transactionManager.getReactiveTransaction(this.transactionDefinition);
+			// This is an around advice: Invoke the next interceptor in the chain.
+			// This will normally result in a target object being invoked.
+			// Need re-wrapping of ReactiveTransaction until we get hold of the exception
+			// through usingWhen.
+			return status.flatMap(it -> Mono.usingWhen(Mono.just(it), ignore -> mono,
```

Comment:
wouldn't it be better to start from `Mono.usingWhen`?
```java
return Mono.usingWhen(status, ignore -> mono, this.transactionManager::commit, ...)
```

Reply:
We require the status value outside of `usingWhen(…)` (see `rollbackOnException`)